dts: Generate GPIO defines in non GPIO devices

erwango opened the pull request:
This pull request aim at generating #define for gpio properties defined in non gpio nodes.
For instance:
```
spbtle-rf@0 {
 	compatible = "st,spbtle-rf";
    	reset-gpios = <&gpioa 8 0>;
};
```
will trigger generation of:
```
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_CONTROLLER	"GPIOA"
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_FLAGS		0
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_PIN		8
```

This is done by introducing a new value in yaml dts bindings attribute "generation": 'define, use-prop-name'. If use-prop-name is set, #define will use property name as cell string instead of property generic name.
To achieve this, I've removed the 'cell-string' yaml attribute. It was used mostly to provide a generic value (IRQ, PINMUX, ..) and only once for a specific case. I enforced the generic value to be used directly in the script to simplify whole generation mechanism.

Last, I updated NXP dts files to use generated _CLOCK_CONTROLLER define.

I also took the opportunity to clean up from the script 'use-name-prefix' which was never used.

This PR is based on #7197

erwango committed:
dts/bindings: Add reset and irq gpios to "st,spbtle-rf" yaml

In order to generate GPIO definition for control pin of BlueNRG module
define reset-gpio and irq-gpio in st,spbtle-rf yaml binding.

Add HAS_DTS_SPI_PINS kconfig symbol to control use of the '#define'

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: disco_l475_iot1/dts: add gpios in bt controller node

Add cs, irq and reset gpios description in spbtle controller
node and use generated defines instead of Kconfig symbols.
Define a bt alias to allow generation of defines with friendly
names.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

GAnthony reviewed (approved):
Since changes to TI dts.fixup files same as in #7197 - this looks good also.

galak commented on dts/bindings/bluetooth/st,spbtle-rf.yaml:
```diff
@@ -18,4 +18,18 @@ properties:
     compatible:
       constraint: "st,spbtle-rf"
 
+    cs-gpios:
```
Hmm, from a device tree binding point of view, this seems to have been done as on the controller not the device side.

MaureenHelm replied:
I don't quite follow. Are you suggesting a particular change? Both st,spbtle-rf and nxp,mcr20a are users of the spi api, which allows specifying a gpio-controlled chip select.

galak replied:
If you look at https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/spi/spi-bus.txt

The cs-gpios property is for the bus controller.  We use the reg property of the spi device as the CS number (ie index in array of cs-gpios - if gpios are used for CS on the controller.)

galak committed:
dts: spi: Add optional cs-gpios to the spi bus controller binding

Now that we can generate gpio info from dts lets add optional support
fot the cs-gpios property in the bus controller node.  This matches the
binding spec from Linux on how cs-gpios are handled.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -84,37 +84,41 @@ config BT_SPI_BLUENRG
 config BT_SPI_CHIP_SELECT_DEV_NAME
 	string "Chip Select (CS) line driver name"
-	depends on BT_SPI_BLUENRG
+	depends on BT_SPI_BLUENRG && !HAS_DTS_SPI_PINS
```
I fixed this up.

galak commented:
recheck
